sematic/ui/src/types/Types.tsx

```diff
@@ -1,899 +1,84 @@
-      variant="outlined"
-      target="blank"
-      endIcon={<OpenInNew />}
-    >
-      <img src={S3Icon} width="20px" style={{paddingRight: "5px"}} alt="S3 icon"/>
-      {s3URI}
-    </Button>
-    </Tooltip>
-  </>;
-}
-
-type ComponentPair = {
-  type: (props: TypeViewProps) => JSX.Element;
-  value: (props: ValueViewProps) => JSX.Element;
+export type ComponentRenderDetails = {
+  value: ExpandViewFunc<AllValueViewProps>;
 };
 
-const TypeComponents: Map<string, ComponentPair> = new Map([
-  ["NoneType", { type: TypeView, value: NoneValueView }],
-  ["float", { type: TypeView, value: FloatValueView }],
-  ["str", { type: TypeView, value: StrValueView }],
-  ["int", { type: TypeView, value: IntValueView }],
-  ["bool", { type: TypeView, value: BoolValueView }],
-  ["FloatInRange", { type: FloatInRangeTypeView, value: FloatValueView }],
-  ["list", { type: ListTypeView, value: ListValueView }],
-  ["tuple", { type: TypeView, value: TupleValueView }],
-  ["dict", { type: TypeView, value: DictValueView }],
-  ["dataclass", { type: DataclassTypeView, value: DataclassValueView }],
-  ["Union", { type: UnionTypeView, value: ValueView }],
-  ["Link", { type: TypeView, value: LinkValueView }],
-  ["datetime.datetime", { type: TypeView, value: DatetimeValueView }],
-  ["enum.Enum", { type: TypeView, value: EnumValueView }],
+// The Registry (fill data)
+const meta: Array<[string, ComponentRenderDetails]> = [
+  ["NoneType", { value: NoneValueView }],
+  ["float", { value: FloatValueView }],
+  ["str", { value: StrValueView }],
+  ["int", { value: IntValueView }],
+  ["bool", { value: BoolValueView }],
+  ["FloatInRange", { value: FloatValueView }],
+  ["list", { value: ListValueView }],
+  ["tuple", { value: TupleValueView }],
+  ["dict", { value: DictValueView }],
+  ["dataclass", { value: DataclassValueView }],
+  ["Union", { value: ValueView }],
+  ["Link", { value: LinkValueView }],
+  ["datetime.datetime", { value: DatetimeValueView }],
+  ["enum.Enum", { value: EnumValueView }],
   [
     "torch.utils.data.dataloader.DataLoader",
-    { type: TypeView, value: TorchDataLoaderValueView },
+    { value: TorchDataLoaderValueView },
   ],
   [
     "plotly.graph_objs._figure.Figure",
-    { type: TypeView, value: PlotlyFigureValueView },
+    { value: PlotlyFigureValueView },
   ],
   [
     "matplotlib.figure.Figure",
-    { type: TypeView, value: MatplotlibFigureValueView },
+    { value: MatplotlibFigureValueView },
   ],
   [
     "pandas.core.frame.DataFrame",
     {
-      type: TypeView,
       value: DataFrameValueView,
     },
   ],
-  [
-    "sematic.types.types.aws.s3.S3Location",
-    {
-      type: TypeView,
-      value: S3LocationValueView,
-    }
-  ],
-  [
-    "sematic.types.types.aws.s3.S3Bucket",
-    {
-      type: TypeView,
-      value: S3BucketValueView,
-    }
-  ]
-]);
+  // (Keep this memo): when a new type is added, see if you need to add a
+  // type presentation type at line `export type AnyTypeRepr =`
```

Comment:
The current type categories should cover all cases, this should not be necessary.

Reply:
I am not an expert on this topic, and I don't know why that is. But I will be sure to follow your suggestion. 

The comment is now removed.